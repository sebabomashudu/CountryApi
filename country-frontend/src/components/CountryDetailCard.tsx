interface CountryDetailCardProps {
  country: {
    name: string;
    flagUrl: string;
    capital: string;
    population: number;
  };
}

export function CountryDetailCard({ country }: CountryDetailCardProps) {
  return (
    <div className="p-4">
      <h1 className="text-3xl font-bold mb-4">{country.name}</h1>
      <img 
        src={country.flagUrl} 
        alt={`${country.name} flag`} 
        className="w-64 h-auto mb-4" 
      />
      <p><strong>Capital:</strong> {country.capital}</p>
      <p><strong>Population:</strong> {country.population.toLocaleString()}</p>
    </div>
  );
}