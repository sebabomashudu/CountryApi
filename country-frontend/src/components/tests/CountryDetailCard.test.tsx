import { render, screen } from '@testing-library/react';
import { CountryDetailCard } from '../CountryDetailCard';
import { describe, it, expect } from 'vitest';
import '@testing-library/jest-dom/vitest';

describe('CountryDetailCard', () => {
  const mockCountry = {
    name: 'Testland',
    flagUrl: 'https://flagcdn.com/test.svg',
    capital: 'Testville',
    population: 8500000
  };

  it('renders the country name and flag', () => {
    render(<CountryDetailCard country={mockCountry} />);

    expect(screen.getByText(mockCountry.name)).toBeInTheDocument();
    expect(screen.getByAltText(`${mockCountry.name} flag`))
      .toHaveAttribute('src', mockCountry.flagUrl);
  });

  it('displays capital information', () => {
    render(<CountryDetailCard country={mockCountry} />);

    expect(screen.getByText(/Capital:/)).toBeInTheDocument();
    expect(screen.getByText(mockCountry.capital)).toBeInTheDocument();
  });

  it('displays formatted population', () => {
    render(<CountryDetailCard country={mockCountry} />);

    expect(screen.getByText(/Population:/)).toBeInTheDocument();
    expect(screen.getByText('8,500,000')).toBeInTheDocument();
  });

  it('matches the snapshot', () => {
    const { container } = render(<CountryDetailCard country={mockCountry} />);
    expect(container).toMatchSnapshot();
  });
});