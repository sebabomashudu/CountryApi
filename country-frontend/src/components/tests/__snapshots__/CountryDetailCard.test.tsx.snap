// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CountryDetailCard > matches the snapshot 1`] = `
<div>
  <div
    class="p-4"
  >
    <h1
      class="text-3xl font-bold mb-4"
    >
      Testland
    </h1>
    <img
      alt="Testland flag"
      class="w-64 h-auto mb-4"
      src="https://flagcdn.com/test.svg"
    />
    <p>
      <strong>
        Capital:
      </strong>
       
      Testville
    </p>
    <p>
      <strong>
        Population:
      </strong>
       
      8,500,000
    </p>
  </div>
</div>
`;
